core/domain/feedback_services.py

```diff
@@ -366,6 +366,34 @@ def _get_thread_from_model(thread_model):
         thread_model.created_on, thread_model.last_updated)
 
 
+def get_messages_for_exp_id(exp_id, limit=None):
```

Comment:
The way this is implemented makes it look like it won't scale well to explorations that have many threads/messages. Is it possible to do something like: query the messages datastore for stuff with exp_id matching the given exp_id, and sort backwards by last_updated, and limit to top 3 or 10 or something? That's going to be more efficient than fetching and iterating through every single message each time.